pkg/bdb/hash_metadata_page.go

```diff
@@ -0,0 +1,47 @@
+package bdb
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/go-restruct/restruct"
+)
+
+// source: https://github.com/berkeleydb/libdb/blob/5b7b02ae052442626af54c176335b67ecc613a30/src/dbinc/db_page.h#L130
+type HashMetadataPage struct {
+	GenericMetadataPage
+	MaxBucket   uint32 `struct:"uint32"` /* 72-75: ID of Maximum bucket in use */
+	HighMask    uint32 `struct:"uint32"` /* 76-79: Modulo mask into table */
+	LowMask     uint32 `struct:"uint32"` /* 80-83: Modulo mask into table lower half */
+	FillFactor  uint32 `struct:"uint32"` /* 84-87: Fill factor */
+	NumKeys     uint32 `struct:"uint32"` /* 88-91: Number of keys in hash table */
+	CharKeyHash uint32 `struct:"uint32"` /* 92-95: Value of hash(CHARKEY) */
+	// don't care about the rest...
+}
+
+func ParseHashMetadataPage(data []byte) (*HashMetadataPage, error) {
+	var metadata HashMetadataPage
+
+	err := restruct.Unpack(data, binary.LittleEndian, &metadata)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unpack HashMetadataPage: %w", err)
+	}
+
+	return &metadata, nil
```

Comment:
We have to call `validate()` here, right? I didn't find where `validate()` is called.

```suggestion
	return &metadata, metadata.validate()
```

Reply:
great catch!